OrbitBase/include/OrbitBase/Tracing.h

```diff
@@ -0,0 +1,62 @@
+#ifndef ORBIT_TRACING_TRACING_H_
+#define ORBIT_TRACING_TRACING_H_
+
+#include <memory>
+
+#define ORBIT_TRACING_ENABLED 1
+
+#if ORBIT_TRACING_ENABLED
+
+// Scoped trace with user defined name.
+#define ORBIT_SCOPE(name) orbit::tracing::Scope ORBIT_UNIQUE(ORB)(name)
+// Scoped trace with calling function name.
+#define ORBIT_SCOPE_FUNC ORBIT_SCOPE(__FUNCTION__)
+// Manual scope begin.
+#define ORBIT_BEGIN(name) ORBIT_CALL(Begin(name))
+// Manual scope end.
+#define ORBIT_END ORBIT_CALL(End())
```

Comment:
For consistency I would have ORBIT_END and ORBIT_SCOPE_FUNC also require parantheses like a function call.